src/ReportManager/SchemaForm/SchemaFormContext/index.js

```diff
@@ -0,0 +1,52 @@
+import React, { createContext } from 'react';
+
+import { FORM_FIELDS_TYPES } from '../constants';
+import { textFieldDetailsFactory } from '../fields/fieldDetailsFactory';
+import { isFieldRequired } from '../utils';
+import {cloneDeep} from "lodash-es";
+
+export const SchemaFormContext = createContext(null);
+
+
+const SchemaFormContextProvider = ({ schema, onFormChange, formData, formErrors, children }) => {
+
+  const onFieldChange = (field, value) => {
+    onFormChange({
+      formData: {
+        [field]: value
+      }
+    });
+  };
+
+  const getFieldDetails = (fieldName) => {
+    const fieldSchema = schema.json.properties[fieldName];
```

Comment:
This also doesn't look like it'll work for collections/nested field data

Reply:
It wont, I'm just setting the grounds, I think any particular implementation/modification should be attended in its own ticket, for this case in the Collection ticket

Reply:
I feel like this is a start, but eventually we can add a util function just like we do in the EFB that translates the entire schema into a flat fields object that makes all the rendering details of each field easy to access, instead of having to call `getFieldDetails` each time, which will require some processing for nested fields and that probably will be called on each render. We can work on that later.